Homework 20

karinamarcial opened the pull request:
Refactored the previous test case deletePlaylist()

karinamarcial committed:
homework16

Navigation to registration

karinamarcial committed:
Add Song To Playlist

homework is done

karinamarcial committed:
addArguments

Added arguments

karinamarcial committed:
Play Song

Play Song test case

karinamarcial committed:
Update HomeWork18.java

Updated homework

karinamarcial committed:
Homework19

Delete playlist

karinamarcial committed:
homework20

Refactored the previous test case deletePlaylist()

Berichicko commented on src/test/java/HomeWork20.java:
```diff
@@ -0,0 +1,40 @@
+
+public class HomeWork20 extends BaseTest {
+    @Test
+    public void deletePlaylist() throws InterruptedException {
+
+        String deletedPlaylistMsg = "Deleted playlist \"karina playlist.\"";
+        provideEmail("karina.usmanova01@testpro.io");
+        providePassword("YrEdlRVe");
+        clickLoginButton();
+
+        clickOnPlaylist();
+        clickDeleteThisPlaylist();
+        Thread.sleep(2000);
```
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

karinamarcial replied:
Good day,
The thing is without  Thread.sleep I am getting an error and my test is failed.
The report shows: java.lang.AssertionError: expected [Deleted playlist "karina playlist."] but found []
It seems like its not able to read a notification pop

Sometimes my code has passed, sometimes test failed. I don't understand why. Screenshots attached. The same code...one time passed, one time failed.
<img width="1710" alt="Screenshot 2024-04-02 at 4 19 38 PM" src="https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/163922153/d033e67d-0012-4fc2-a861-81467f4539ce">
<img width="1710" alt="Screenshot 2024-04-02 at 4 20 26 PM" src="https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/163922153/dadae46d-d8fd-4476-bcfe-058d77f6020c">

karinamarcial replied:
Can you please overview my code. Maybe I need to use different method in getDeletedPlaylistSuccessMsg? not this one: .presenceOfElementLocated

karinamarcial commented:
Good day, Alex

 Without Thread.sleep I am getting an error and my test is failed.
The report shows: java.lang.AssertionError: expected [Deleted playlist "karina playlist."] but found []
It seems like its not able to read a notification pop

Sometimes my code has passed, sometimes test failed. I don't understand why. Screenshots attached. The same code...one time passed, one time failed.

Can you please overview my code. Maybe I need to use different method in getDeletedPlaylistSuccessMsg? not this one: .presenceOfElementLocated

________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Tuesday, April 2, 2024 1:30 AM
To: Test-Pro-Bootcamp/A57-QA-Automation-Framework ***@***.***>
Cc: Karina Usmanova ***@***.***>; Author ***@***.***>
Subject: Re: [Test-Pro-Bootcamp/A57-QA-Automation-Framework] Homework 20 (PR #56)


@Berichicko commented on this pull request.

________________________________

In src/test/java/HomeWork20.java<https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/pull/56#discussion_r1547159304>:

> +    public void deletePlaylist() throws InterruptedException {
+
+        String deletedPlaylistMsg = "Deleted playlist \"karina playlist.\"";
+        ***@***.***");
+        providePassword("YrEdlRVe");
+        clickLoginButton();
+
+        clickOnPlaylist();
+        clickDeleteThisPlaylist();
+        Thread.sleep(2000);


Please, delete throws InterruptedException

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

—
Reply to this email directly, view it on GitHub<https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/pull/56#pullrequestreview-1972644800>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BHCUB2OT7GR7BNI3DUGHUQLY3I66DAVCNFSM6AAAAABFSFS2YGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNZSGY2DIOBQGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

karinamarcial committed:
homework20new

new

karinamarcial committed:
Update HomeWork20.java

finally found out solution. i used the visibility condition instead of presence

Berichicko commented:
Hi dear Karina

I ran the tests 10 times and all tests passed

![image](https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/104761482/852e9fb8-027e-4e14-9344-3529727356e8)


![image](https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/104761482/a3af7561-be33-427e-9944-2587ccb203a7)


HINT

1) Try CSS selector

   `WebElement notification = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));`
   
 2) When running automated tests with Selenium, it's often done without a visible browser window (headless mode). Moving the mouse during these tests can cause unexpected behavior because mouse events might interfere with elements on the web page. Therefore, it's recommended to avoid mouse interaction and instead use Selenium's API to control elements on the page, like clicks and input.((do not touch the physical mouse during tests)
